### PR TITLE
[FIX] stock_picking_batch: remove empty pickings

### PR DESCRIPTION
**Pull request by adwid**
This is an adaptation of [1] (which is already deployed in 15.0)

[1] eb0ebc077afed24e8f5c2212f2a216bb1b8e6936

OPW-2711585

### CODE REVIEW

**Comment by adwid** [comment]
@fw-bot up to 14.0

File: addons/stock_picking_batch/models/stock_picking_batch.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def done(self):
                 picking_to_backorder |= picking
             else:
                 picking.action_done()
-        if picking_without_qty_done:
+        if picking_without_qty_done == pickings:
```
Comment (ticodoo) [inline addons/stock_picking_batch/models/stock_picking_batch.py:86]: Not really a big deal, but `len(picking_without_qty_done) == len(pickings)` is slightly more performant.

**Review by ticodoo — APPROVED** [review]
LGTM, my comments are minor improvements that aren't required. I would add more description to the commit message though since it's not very obvious what is being fixed. E.g.:

When validating a batch picking where some pickings have no quantities done + some do, then avoid immediate transfers wizard and remove pickings with no quantities done from the batch picking. This will make it so users can validate a partially done batch picking in this case.

**Comment by adwid** [comment]
I have updated the code and the commit description (I used your suggestion, thx for this :ok_hand: )

**Comment by ticodoo** [comment]
@robodoo r+

**Comment by adwid** [comment]
Yep, sorry for that
The correction is being merged (#82228)

**Comment by JKE-be** [comment]
Thank you for the fix.
But my messages what not especially about my dbs ... I just reverted the code meanwhile ;)
But to show you again the risk to break a version for more users that the use case that you fix.
The purist will say, no traceback no bug...
To be more tolerant: no other way to do it without a fix and not a new feature..., need to check the cost feature/risk...

Here it is imo feature, and risk > case fixed.
You can always create an automatic action to handle backorder if needed or ... 
And your backport risk to break other usecases...

E.g. mine ;)
Send all picking ready in a new picking.batch to the logistician, close my batch that will close all picking and mark qty done to qty ordered and finally close my batch if no error... 
-> if not closed: error of stock to handle manually... (check if error in our stock, error in stock of logistician, if parcel is sent or not and so create backorder if needed ...) and not especially automatically generate back order... So I will perso keep the revert to be safe ;)

**Pull request closed without merging**